features/src/main/scala/com/salesforce/op/stages/base/sequence/BinarySequenceEstimator.scala

```diff
@@ -0,0 +1,155 @@
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.stages.base.sequence
+
+import com.salesforce.op.features.FeatureSparkTypes
+import com.salesforce.op.features.types.{FeatureType, FeatureTypeSparkConverter}
+import com.salesforce.op.stages._
+import org.apache.spark.ml.{Estimator, Model}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{Dataset, Encoder, Encoders}
+import org.apache.spark.util.ClosureUtils
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.runtime.universe.TypeTag
+import scala.util.Try
+
+/**
+ * Takes an input feature of type I1 and a sequence of input features of type I2 and performs
+ * a fit operation in order to define a transformation for those (or similar) features. This
+ * abstract class should be extended when settable parameters are needed within
+ * the fit function
+ *
+ * @param operationName unique name of the operation this stage performs
+ * @param uid           uid for instance
+ * @param tti1          type tag for input1
+ * @param tti2          type tag for input2
+ * @param tto           type tag for input
+ * @param tti1v         type tag for input1 value
+ * @param tti2v         type tag for input2 value
+ * @param ttov          type tag for output value
+ * @tparam I1 input single feature type
+ * @tparam I2 input sequence feature type
+ * @tparam O output feature type
+ */
+abstract class BinarySequenceEstimator[I1 <: FeatureType, I2 <: FeatureType, O <: FeatureType]
+(
+  val operationName: String,
+  val uid: String
+)(
+  implicit val tti1: TypeTag[I1],
+  val tti2: TypeTag[I2],
+  val tto: TypeTag[O],
+  val tti1v: TypeTag[I1#Value],
```

Comment:
please rename `tti1v` -> `ttiv1` and `tti2v` -> `ttiv2` (since we have those in other classes as well) + update the docs